src/MerchantFrontend/src/app/core/services/merchant-api.service.ts

```diff
@@ -1,65 +1,128 @@
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-import { HttpClient } from '@angular/common/http';
-import { Merchant, MerchantRegistration } from '../models/merchant.model';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Merchant, MerchantLoginResult, MerchantRegistration } from '../models/merchant.model';
 import { Product } from '../models/product.model';
 import { Authority } from '../models/authority.model';
+import { StateService } from '../state/state.service';
+import { map, take } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class MerchantApiService {
 
-    private apiUrl ='';
+  private merchantSwAccessKey = 'SWSCJTWTSBF_G77_ML-5KRPWTA';
+  private apiUrl = 'http://localhost:8888';
 
-  constructor(private readonly http: HttpClient) { }
+  constructor(
+    private readonly http: HttpClient,
+    private stateService: StateService
+  ) { }
 
-  login(mail: string, password: string): Observable<any> {
-      const body = JSON.stringify(
-          {
-              'email': mail,
-              'password': password,
-          }
+  login(username: string, password: string): Observable<MerchantLoginResult> {
 
-      );
-      return this.http.post(this.apiUrl + '/merchant-api/login', body);
+    let headers = new HttpHeaders();
+    headers = headers.set('Content-Type', 'application/json');
+    headers = headers.set('sw-access-key', this.merchantSwAccessKey);
+
+    const body = JSON.stringify(
+      {
+        username: username,
+        password: password,
+      }
+    );
+    return this.http.post<MerchantLoginResult>(this.apiUrl + '/sales-channel-api/v1/customer/login', body, {headers: headers});
   }
 
   // merchant routes
 
-  registerMerchant(merchantRegistration: MerchantRegistration): Observable<Merchant> {
-      return this.http.post<Merchant>(this.apiUrl + '/merchant-api/register', JSON.stringify(merchantRegistration));
+  registerMerchant(merchantRegistration: MerchantRegistration): Observable<any> {
+      return this.http.post<any>(this.apiUrl + '/merchant-api/register', JSON.stringify(merchantRegistration))
   }
 
   getMerchant(): Observable<Merchant> {
-      return this.http.get<Merchant>(this.apiUrl + '/merchant-api/register');
+    let headers = new HttpHeaders();
+    headers = headers.set('Content-Type', 'application/json');
+    headers = headers.set('sw-access-key', this.merchantSwAccessKey);
+    headers = headers.set('sw-context-token', this.getSwContextToken());
+
+    return this.http.get<any>(this.apiUrl + '/sales-channel-api/v1/customer', {headers: headers})
```

Comment:
This should be changed to `/merchant-api/profile` tomorrow - I will notify you, when this is merged